src/kmeans.jl

```diff
@@ -54,13 +54,27 @@ function kmeans!(X::AbstractMatrix{<:Real},                # in: data matrix (d
     dc, k = size(centers)
 
     d == dc || throw(DimensionMismatch("Inconsistent array dimensions for `X` and `centers`."))
-    (2 <= k < n) || throw(ArgumentError("k must have 2 <= k < n=$n ($k given)."))
-    if weights !== nothing
-        length(weights) == n || throw(DimensionMismatch("Incorrect length of weights."))
-    end
+    (1 <= k <= n) || throw(ArgumentError("k must have 1 <= k <= n=$n ($k given)."))
```

Comment:
Let's take the opportunity to improve the message:
```suggestion
    (1 <= k <= n) || throw(ArgumentError("k must be from 1:n (n=$n), k=$k given."))
```